### PR TITLE
Adds Scala as a community SDK

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?

Temporal [deprecated the Scala repo](https://github.com/temporalio/sdk-scala). Per [this discussion](https://github.com/temporalio/sdk-java/issues/1007), a community member added Scala support for Workflows. 

> It allows you to define and use workflows in a Scala way!

[ZIO Temporal](https://github.com/vitaliihonta/zio-temporal)

## Notes to reviewers

Also reordered the SDKs alphabetically. 

### Discussion

Should we remove the community Python SDK - since we have a Python SDK in beta - or leave it since this page talks about communities SDKs?

### CODE REVIEW

File: docs/external-resources.md [hunk 1]
```diff
@@ -155,8 +155,9 @@ _Note: Each of [our SDK docs](/application-development) have a Hello World and R
 
 ## Community SDKs
```
Comment (flossypurse) [inline docs/external-resources.md:156]: @rachfop Let's just move this list to this section https://docs.temporal.io/temporal#third-party-sdks and remove this from external resources.
Long game is to break up the content on this page, redistribute it, and remove this page.

File: docs/concepts/what-is-a-temporal-sdk.md [hunk 2]
```diff
@@ -38,4 +38,6 @@ The following SDKs are in alpha/pre-alpha development stages, but are not yet su
 
 The following third party SDKs exist, but are not supported in the application development guide:
```
Comment (djmagee) [inline docs/concepts/what-is-a-temporal-sdk.md:39]:
```suggestion
The following third-party SDKs exist but are not supported in the Application development guide:
```

**Review by djmagee — APPROVED** [review]

File: docs/concepts/what-is-a-temporal-sdk.md [hunk 3]
```diff
@@ -36,6 +36,8 @@ The following SDKs are in alpha/pre-alpha development stages, but are not yet su
 
 ### Third-party SDKs
 
-The following third party SDKs exist, but are not supported in the application development guide:
+The following third-party SDKs exist but are not supported in the Application development guide:
```
Comment (flossypurse) [inline docs/concepts/what-is-a-temporal-sdk.md:39]:
```suggestion
The following third-party SDKs exist but are not supported in the [Application development guide](/application-development):
```

**Review by flossypurse — APPROVED** [review]

File: docs/concepts/what-is-a-temporal-sdk.md [hunk 4]
```diff
@@ -36,6 +36,8 @@ The following SDKs are in alpha/pre-alpha development stages, but are not yet su
 
 ### Third-party SDKs
 
-The following third party SDKs exist, but are not supported in the application development guide:
+The following third-party SDKs exist but are not supported in the [Application development guide](/application-development):
 
-- [coinbase/temporal-ruby](https://github.com/coinbase/temporal-ruby)
+- [Scala](https://github.com/vitaliihonta/zio-temporal) from [@vitaliihonta](https://github.com/vitaliihonta)
+- [Python](https://github.com/firdaus/temporal-python-sdk) from [@firdaus](https://github.com/firdaus)
```
Comment (bergundy) [inline docs/concepts/what-is-a-temporal-sdk.md:42]: This one isn’t maintained afaik and I don’t see a reason to mention it at this point

File: docs/external-resources.md [hunk 5]
```diff
@@ -153,11 +153,6 @@ _Note: Each of [our SDK docs](/application-development) have a Hello World and R
 - [DataDog/temporalite](https://github.com/DataDog/temporalite) - Single binary distro for CI (see talk)
```
Comment (bergundy) [inline docs/external-resources.md:153]: Might as well update this, it’s now in our org

**Pull request merged**